Analyse_Birds.qmd

```diff
@@ -107,13 +122,220 @@ required_packages <- c(
+
+data_carroll <- data_carroll %>%
+  mutate(longitude_column = -89.9826,
+         latitude_column = 42.0956)
+
+data_lafayette <- data_lafayette %>%
+  mutate(longitude_column = -90.1160,
+         latitude_column = 42.6825)
+
+data_iowa <- data_iowa %>%
+  mutate(longitude_column = -90.1160,
+         latitude_column = 42.6825)
+
+data_dane <- data_dane %>%
+  mutate(longitude_column = -89.4012,
+         latitude_column = 43.0731)
+
+data_green <- data_green %>%
+  mutate(longitude_column = -89.6387,
+         latitude_column = 42.6017)
+
+data_grant <- data_grant %>%
+  mutate(longitude_column = -90.7082,
+         latitude_column = 42.8475)
+
+data_rock <- data_rock %>%
+  mutate(longitude_column = -89.0187,
+         latitude_column = 42.6828)
+
+data_monroe <- data_monroe %>%
+  mutate(longitude_column = -90.8129,
+         latitude_column = 43.9447)
+
+data_lee_ogle <- data_lee_ogle %>%
+  mutate(longitude_column = -89.4038,
+         latitude_column = 41.9293)
+
+data_carroll_whiteside <- data_carroll_whiteside %>%
+  mutate(longitude_column = -89.9742,
+         latitude_column = 41.9527)
+```
+
+Now, for each county dataset, we subdivide the said dataset for each year when at least one nest has been observed in this county. 
+
+```{r}
+county_names <- c("data_will", "data_jo_daviess", "data_lafayette", "data_iowa", 
+                  "data_dane", "data_green", "data_grant", "data_grundy", 
+                  "data_lee_ogle", "data_lee", "data_rock", "data_monroe", 
+                  "data_carroll", "data_carroll_whiteside")
+
+county_dataframes <- mget(county_names)
+tables <- c()
+
+for(i in seq_along(county_dataframes)) {
+    county <- county_dataframes[[i]]
+    county_name <- county_names[i]
+    
+    if (!is.data.frame(county)) {
+        print(paste0("Error: ", county_name, " is not a dataframe"))
+        next
+    }
+    
+    if (!"Year" %in% colnames(county)) {
+        print(paste0("Error: ", county_name, " does not contain a 'Year' column"))
+        next
+    }
+    
+    data_county_year <- split(county, county$Year)
+
+    for(year in min(county$Year):max(county$Year)){
+        tableau <- data_county_year[[as.character(year)]]
+        
+        if (!is.null(tableau)) {
+            variable_name <- paste0(county_name, "_", year)
+            tables <- c(tables, variable_name)
+            assign(variable_name, tableau, envir = .GlobalEnv)
+        }
+    }
+}
+```
+
+We went from a full dataset with everything mixed-up to multiple datasets for each year and each county when an observation is available. 
+
+Here are, for example, the first rows of the dataframe related to the county of Will in 1995
+
+```{r}
+head(data_will_1995)
+```
+
+The main information that we want to present on the map are the mean number of fledged birds by nest and the amount of data available for each county and each year. 
+
+Here is the creation of a summary dataframe that contains all theese information for each dataframe. 
```

Comment:
these

Reply:
The typo is now corrected, thank you for spotting it.